ignite/contrib/handlers/base_logger.py

```diff
@@ -172,17 +172,21 @@ def attach(
         Returns:
             :class:`~ignite.base.events.RemovableEventHandle`, which can be used to remove the handler.
         """
+        # create a list of the available events
+        state_events = []
+        for event in State.attr_to_events.values():
+            state_events.extend(event)
+
         if isinstance(event_name, EventsList):
             for name in event_name:
-                if name not in State.event_to_attr:
+                if name not in state_events:
```

Comment:
I think you have to dig more into the code to understand why we did that previously and thus decide how to make a proper code change